eodag/plugins/authentication/aws_auth.py

```diff
@@ -283,3 +285,38 @@ def get_rio_env(self) -> dict[str, Any]:
             "session": self.s3_session,
             **rio_env_kwargs,
         }
+
+    def presign_url(
+        self,
+        asset: Asset,
+        expires_in: int = 3600,
+    ) -> str:
+        """This method is used to presign a url to download an asset from S3.
+
+        :param asset: asset for which the url shall be presigned
+        :param expires_in: expiration time of the presigned url in seconds
+        :returns: presigned url
+        :raises: :class:`~eodag.utils.exceptions.EodagError`
+        :raises: :class:`NotImplementedError`
+        """
+        if not getattr(self.config, "support_presign_url", True):
+            raise NotImplementedError(
+                f"presign_url is not supported for provider {self.provider}"
+            )
+
+        url_parts = urlparse(asset["href"])
+
+        s3_client = self.get_s3_client()
+        url_path_parts = url_parts.path[1:].split("/")  # remove leading "/" and split
```

Comment:
Here we should reuse the existing method we already have to do this: `get_bucket_name_and_prefix` from `eodag/utils/__init__.py`.

It is already used in `get_product_bucket_name_and_prefix` from `eodag/plugins/download/aws.py`